test/alphabet/nucleotide/dna4_test.cpp

```diff
@@ -0,0 +1,327 @@
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <gtest/gtest.h>
+#include <sstream>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+// default/zero construction
+TEST(dna4, ctr)
+{
+    dna4 t1;
+}
+
+// zero initialization
+TEST(dna4, zro)
+{
+    dna4 t0{};
+    EXPECT_EQ(t0, dna4::A);
+}
+
+// copy construction
+TEST(dna4, cp_ctr)
+{
+    dna4 t1{dna4::C};
+    dna4 t2{t1};
+    dna4 t3(t1);
+    EXPECT_EQ(t1, t2);
+    EXPECT_EQ(t2, t3);
+}
+
+// move construction
+TEST(dna4, mv_ctr)
+{
+    dna4 t0{dna4::C};
+    dna4 t1{dna4::C};
+    dna4 t2{std::move(t1)};
+    EXPECT_EQ(t2, t0);
+    dna4 t3(std::move(t2));
+    EXPECT_EQ(t3, t0);
+}
+
+// copy assignment
+TEST(dna4, cp_assgn)
+{
+    dna4 t1{dna4::C};
+    dna4 t2;
+    dna4 t3;
+
+    t2 = t1;
+    t3 = t1;
+    EXPECT_EQ(t1, t2);
+    EXPECT_EQ(t2, t3);
+}
+
+// move assignment
+TEST(dna4, mv_assgn)
+{
+    dna4 t0{dna4::C};
+    dna4 t1{dna4::C};
+    dna4 t2;
+    dna4 t3;
+    t2 = std::move(t1);
+    EXPECT_EQ(t2, t0);
+    t3 = std::move(t2);
+    EXPECT_EQ(t3, t0);
+}
+
+// swap
+TEST(dna4, swap)
+{
+    dna4 t0{dna4::C};
+    dna4 t1{dna4::C};
+    dna4 t2{};
+    dna4 t3{};
+
+    std::swap(t1, t2);
+    EXPECT_EQ(t2, t0);
+    EXPECT_EQ(t1, t3);
+}
+
+// comparison
+TEST(dna4, cmp)
+{
+    dna4 t0{dna4::A};
+    dna4 t1{dna4::C};
+    dna4 t2{dna4::G};
+
+    EXPECT_LT(t0, t1);
+    EXPECT_LE(t0, t1);
+    EXPECT_LE(t1, t1);
+    EXPECT_EQ(t1, t1);
+    EXPECT_GE(t1, t1);
+    EXPECT_GE(t2, t1);
+    EXPECT_GT(t2, t1);
+}
+
+TEST(dna4, to_char_member)
+{
+    EXPECT_EQ(dna4::A.to_char(), 'A');
+    EXPECT_EQ(dna4::C.to_char(), 'C');
+    EXPECT_EQ(dna4::G.to_char(), 'G');
+    EXPECT_EQ(dna4::T.to_char(), 'T');
+    EXPECT_EQ(dna4::U.to_char(), 'T');
+    EXPECT_EQ(dna4::UNKNOWN.to_char(), 'A');
+}
+
+TEST(dna4, to_char_free)
```

Comment:
Hm, it's part of the alphabet concept and thereby part of the "official interface". I don't think it hurts to test it here.